### PR TITLE
[WIP] remove chainec

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

File: chaincfg/chainec/chainec.go [hunk 1]
```diff
@@ -38,7 +40,7 @@ type PublicKey interface {
 	GetY() *big.Int
 
 	// GetType returns the ECDSA type of this key.
-	GetType() int
+	GetType() dcrec.SignatureType
```
Comment (dnldd) [inline chaincfg/chainec/chainec.go]: The interfaces here can be moved into `dcrec/common.go`, coupled with the other changes suggested `the chainec` package can then be removed.

File: chaincfg/chainec/edwards.go [hunk 2]
```diff
@@ -278,7 +279,7 @@ func newEdwardsDSA() DSA {
 			return edwards.GenerateKey(edwardsCurve, rand)
 		},
 		sign: func(priv PrivateKey, hash []byte) (r, s *big.Int, err error) {
-			if priv.GetType() != ECTypeEdwards {
+			if priv.GetType() != dcrec.ECTypeEdwards {
```
Comment (dnldd) [inline chaincfg/chainec/edwards.go]: `edwardsDSA` is a wrapper for the underlining type. I think we're better off removing this and using the underlining type directly.

File: chaincfg/chainec/secp256k1.go [hunk 3]
```diff
@@ -280,7 +281,7 @@ func newSecp256k1DSA() DSA {
 			return secp256k1.GenerateKey(rand)
 		},
 		sign: func(priv PrivateKey, hash []byte) (r, s *big.Int, err error) {
-			if priv.GetType() != ECTypeSecp256k1 {
+			if priv.GetType() != dcrec.ECTypeSecp256k1 {
```
Comment (dnldd) [inline chaincfg/chainec/secp256k1.go]: same with `secp256k1DSA`, better off using the underlining type directly.

**Comment by dajohi** [comment]
Closed in favor of #1304

**Pull request closed without merging**